lib/async/enumerable.rb

```diff
@@ -25,12 +26,18 @@ module Enumerable
     DEFAULT_MAX_FIBERS = 1024
```

Comment:
since this is now defined in the config class, we should defer the responsibility of knowing and giving default_max_fibers to the config class. -- that means this class should only really worry about having a config and just using whatever the config has for max_fibers rather than trying to discern/manage what the default is